src/ExtensionConfigHandler.php

```diff
@@ -352,6 +352,156 @@ protected function getSourceStorageWrapper(array $source_dirs, $subdir = Install
+      if ($dependencies[$dependency_type]) {
+        sort($dependencies[$dependency_type]);
+        $dependencies[$dependency_type] = array_values($dependencies[$dependency_type]);
+      }
+      else {
+        unset($dependencies[$dependency_type]);
+      }
+    }
+    return $dependencies;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfigDependancies($extension, $limit = NULL) {
+    $dependencies = array();
+    $listed_config = $this->getExtensionInfo($extension, 'managed');
+    foreach ($listed_config as $key) {
+      $config_dependencies = $this->getIndividualConfigDependancies($key, $limit);
+      foreach ($config_dependencies as $dependency_type => $type_dependencies) {
+        if (!isset($dependencies[$dependency_type])) {
+          $dependencies[$dependency_type] = array();
+        }
+        $dependencies[$dependency_type] += $type_dependencies;
+      }
+    }
+    // Exclude 'core' and the specified extension from the full list.
+    unset($dependencies['module']['core']);
+    unset($dependencies['module'][$extension]);
+    return $this->sortAndFilterOutput($dependencies);
+  }
+
+  /**
+   * @param $name
+   *   The config key to find the dependencies of.
+   * @param null $recursion_limit
+   *   Optionally limit the levels of recursion.
+   * @return mixed
+   */
+  public function getIndividualConfigDependancies($name, $recursion_limit = NULL) {
+    static $recursive_iterations = 0;
+    static $checked = array();
+
+    if (!isset($checked[$name])) {
+      $config_dependencies = \Drupal::config($name)->get('dependencies');
+      if ($config_dependencies && is_array($config_dependencies)) {
+        foreach ($config_dependencies as $dependency_type => $type_dependencies) {
+          // Use associative array to avoid duplicates.
+          $config_dependencies[$dependency_type] = array_combine($type_dependencies, $type_dependencies);
+        }
+
+        // Recurse to find sub-dependencies.
+        if (isset($config_dependencies['config'])) {
+          $recursive_iterations++;
+          if ($recursion_limit && $recursive_iterations < $recursion_limit) {
+            foreach ($config_dependencies['config'] as $dependency) {
+              $sub_dependencies = $this->getIndividualConfigDependancies($dependency, $recursion_limit);
+
+              // Add this dependency's dependencies to the list to be returned.
+              foreach ($sub_dependencies as $dependency_type => $type_dependencies) {
+                if (!isset($config_dependencies[$dependency_type])) {
+                  $config_dependencies[$dependency_type] = array();
+                }
+                $config_dependencies[$dependency_type] += $type_dependencies;
+              }
+            }
+          }
+          $recursive_iterations--;
+        }
+      }
+      else {
+        $config_dependencies = array();
+      }
+
+      // Config provider is an implied module dependency.
```

Comment:
I don't know how easy it is to establish, but there's a chance that the config could be from a theme! Maybe a call to getExtensionType is needed? (If that returned 'profile' or anything other than 'theme', then it would be fine to use 'module' here)